### PR TITLE
dumpling: update and refine dumpling doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

1. Delete deprecated mydumper from migrate instruction.
2. Add how to control memory usage for tidb and `--params` parameter for dumpling.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4857
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [x] Delete files
- [x] Change aliases
- [ ] Have version specific changes 
- [x] Might cause conflicts

### CODE REVIEW

**Review by yikeke — APPROVED** [review]
Rest LGTM

File: dumpling-overview.md [hunk 1]
```diff
@@ -221,6 +221,14 @@ The `--snapshot` option can be set to a TSO (the `Position` field output by the
 
 The TiDB historical data snapshots when the TSO is `417773951312461825` and the time is `2020-07-02 17:12:45` are exported.
 
+### Control the memory usage of exporting large tables
+
+When Dumpling is exporting a large single table from TiDB, Out of Memory (OOM) might occur because the exported data size is too large, which causes connection abort and export failure. You can use the following parameters to reduce the memory usage of TiDB:
+
++ Setting `--row` to split the data to be exported into chunks. This reduces the memory overhead of TiDB's data scan and enables concurrent table data dump to improve export efficiency.
```
Comment (lichunzhu) [inline dumpling-overview.md:228]: The parameter in dumpling is `--rows`. Besides, what's the difference between the translation of `表内并发` and `表间并发` in English?
```
+ Setting `--rows` to split the data to be exported into chunks. This reduces the memory overhead of TiDB's data scan and enables concurrent table data dump to improve export efficiency.
```

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: dumpling-overview.md (hunk 1 above)
Comment (TomShawn) [inline dumpling-overview.md:228]:
```suggestion
+ Setting `--rows` to split the data to be exported into chunks. This reduces the memory overhead of TiDB's data scan and enables concurrent table data dump to improve export efficiency.
```

Comment (TomShawn) [reply dumpling-overview.md]: The parameter corrected in https://github.com/pingcap/docs/pull/4236/commits/b63e070dd722d3937e4873f07566f4edff6fcc30. Could you explain a little bit what `表间并发` is? I only searched 2 results in Google. 
image

Comment (lichunzhu) [reply dumpling-overview.md]: `表间并发` means that different tables can be dumped parallelly by `--threads` threads. Even without `-r` argument dumpling will dump tables in this way. When `--rows` is enabled, dumpling's single table will be split into several chunks and dumped parallelly.

I understand. `concurrent table data dump` LGTM.

**Pull request merged**